virttest/utils_misc.py

```diff
@@ -2755,14 +2755,26 @@ def format_linux_disk(session, did, mountpoint, size, fstype="ext3"):
     for disk in disks.splitlines():
         if did in disk:
             kname = disk.split()[0]
+            list_parttable_cmd = "parted -s %s print" % kname
+            output = session.cmd_output(list_parttable_cmd)
+            if "Error" in output:
+                logging.info("Create partition table on disk '%s'" % kname)
+                mklabel_cmd = "parted -s %s mklabel msdos" % kname
+                session.cmd(mklabel_cmd)
             logging.info("Create partition on disk '%s'" % kname)
             mkpart_cmd = "parted -s %s  mkpart primary 0 %s" % (kname, size)
             session.cmd(mkpart_cmd)
+            time.sleep(2)
```

Comment:
This line can be removed because you have informed the OS of partition table changes via `partprobe` below.